jsx/FilterableDataTable.js

```diff
@@ -48,18 +52,48 @@ class FilterableDataTable extends Component {
     history.replaceState({}, '', '?');
   }
 
+  /**
+   * Returns the filter state, with filters that are
+   * set to an invalid option removed from the returned
+   * filters
+   *
+   * @return {object}
+   */
+  validFilters() {
+      let filters = {};
+      this.props.fields.forEach((field) => {
+        const filtername = field.filter.name;
+        const filterval = this.state.filter[filtername];
+        if (!this.state.filter[filtername]) {
+            return;
+        }
+
+        if (field.filter.type !== 'select') {
+            filters[filtername] = filterval;
+            return;
+        }
+
+        if (!(filterval.value in field.filter.options)) {
```

Comment:
Would it perhaps be better to clear the filter from the state if it has become invalid?

Reply:
I don't think so. In the situation where filter B depends on filter A. If A changes so that B is invalid, and then nothing happens and A changes back to its original state, you want B to also go back to its original state.